pkg/custmsg/logger.go

```diff
@@ -3,28 +3,37 @@ package custmsg
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 )
 
-const LogLevelKey = "log_level"
+const (
+	LogLevelKey                  = "log_level"
+	defaultBeholderEmitTimeoutMs = 100
+)
 
 type beholderLogger struct {
-	underlying logger.Logger
-	emitter    MessageEmitter
+	underlying  logger.Logger
+	emitter     MessageEmitter
+	emitTimeout time.Duration
 }
 
-func NewBeholderLogger(underlying logger.Logger, emitter MessageEmitter) *beholderLogger {
+func NewBeholderLogger(underlying logger.Logger, emitter MessageEmitter, emitTimeout time.Duration) *beholderLogger {
+	if emitTimeout == 0 {
+		emitTimeout = defaultBeholderEmitTimeoutMs * time.Millisecond
+	}
 	return &beholderLogger{
```

Comment:
Another option is to export the field for the caller to set themselves:
```go
type BeholderLogger struct {
	underlying  logger.Logger
	emitter     MessageEmitter
	EmitTimeout time.Duration
}

func NewBeholderLogger(underlying logger.Logger, emitter MessageEmitter) *BeholderLogger {
	return &beholderLogger{
```
Does introduce a bit of risk though since they need to set it before they start reading it concurrently, and then can't mutate it after that point.

Reply:
nbd but isn't it injected as an interface? This would only be callable on the returned exported type.

Reply:
Another option that doesn't change this signature is a `WithEmitTimeout` method. That would be a safe way to make it available downstream too.

Reply:
@jmank88 done